grovedb/src/tests.rs

```diff
@@ -239,3 +239,62 @@ fn test_root_tree_leafs_are_noted() {
     assert_eq!(db.root_leaf_keys, hm);
     assert_eq!(db.root_tree.leaves_len(), 2);
 }
+
+#[test]
+fn test_checkpoint() {
+    let mut db = make_grovedb();
+    let element1 = Element::Item(b"ayy".to_vec());
+
+    db.insert(&[], b"key1".to_vec(), Element::empty_tree())
+        .expect("cannot insert a subtree 1 into GroveDB");
+    db.insert(&[b"key1"], b"key2".to_vec(), Element::empty_tree())
+        .expect("cannot insert a subtree 2 into GroveDB");
+    db.insert(&[b"key1", b"key2"], b"key3".to_vec(), element1.clone())
+        .expect("cannot insert an item into GroveDB");
+
+    assert_eq!(
+        db.get(&[b"key1", b"key2"], b"key3")
+            .expect("cannot get from grovedb"),
+        element1
+    );
+
+    let checkpoint_tempdir = TempDir::new("checkpoint").expect("cannot open tempdir");
+    let mut checkpoint = db
+        .checkpoint(checkpoint_tempdir.path().join("checkpoint"))
+        .expect("cannot create a checkpoint");
+
+    assert_eq!(
+        db.get(&[b"key1", b"key2"], b"key3")
+            .expect("cannot get from grovedb"),
+        element1
+    );
+    assert_eq!(
+        checkpoint
+            .get(&[b"key1", b"key2"], b"key3")
+            .expect("cannot get from checkpoint"),
+        element1
+    );
+
+    let element2 = Element::Item(b"ayy2".to_vec());
+    let element3 = Element::Item(b"ayy3".to_vec());
+
+    checkpoint
+        .insert(&[b"key1"], b"key4".to_vec(), element2.clone())
+        .expect("cannot insert into checkpoint");
+
+    db.insert(&[b"key1"], b"key4".to_vec(), element3.clone())
+        .expect("cannot insert into GroveDB");
+
+    assert_eq!(
+        checkpoint
+            .get(&[b"key1"], b"key4")
+            .expect("cannot get from checkpoint"),
+        element2,
+    );
+
+    assert_eq!(
+        db.get(&[b"key1"], b"key4")
+            .expect("cannot get from GroveDB"),
+        element3
+    );
```

Comment:
Let's just add an assert that verifying that a key/value pair inserted into the checkpoint doesn't appear in the db and vice versa.